### PR TITLE
Remove deprecated SingleStateFeaturizer subclasses and method.

### PR DESCRIPTION
**Pull request by kedz**
**Proposed changes**:
Removes deprecated `SingleStateFeaturizer` subclasses: `BinarySingleStateFeaturizer` and `LabelTokenizerSingleStateFeaturizer`. Deprecated method `encode_all_actions` of `SingleStateFeaturizer` has been deleted.

No links need to be removed from the docs/changelog as these components did not have sections in the docs (that weren't autogenerated from the class definitions).

They also did not have any tests (!) so tests remain unchanged.

**Status (please check what you already did)**:
- [x] deprecated code has been removed.
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by kedz** [comment]
/modeltest

```yml
include:
 - dataset: ["financial-demo", "Sara"]
   config: ["all-core"]
```

**Comment by kedz** [comment]
/modeltest

```yml
include:
 - dataset: ["helpdesk-assistant"]
   config: ["all-core"]
```

File: changelog/8879.removal.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Follow through on removing deprecated dialogue policy state featurizers: `BinarySingleStateFeature` and `LabelTokenizerSingleStateFeaturizer`.
```
Comment (samsucik) [inline changelog/8879.removal.md:1]: In the interest of simple and kind of consistent language across our changelog:
```suggestion
Removed the deprecated dialogue policy state featurizers: `BinarySingleStateFeature` and `LabelTokenizerSingleStateFeaturizer`.
```

File: changelog/8879.removal.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+
+Remove deprecated `SingleStateFeaturizer` method `encode_all_actions`. Use `encode_all_labels` instead.
```
Comment (samsucik) [inline changelog/8879.removal.md:3]:
```suggestion
Removed the deprecated method `encode_all_actions` of `SingleStateFeaturizer`. Use `encode_all_labels` instead.
```

**Review by samsucik — APPROVED** [review]
Lgtm, let's get it merged 🙂

**Pull request merged**